src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1371,7 +1371,7 @@ private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(Implicit
                     if (types.Any(t => t is INamedTypeSymbol))
                     {
                         return types.OfType<INamedTypeSymbol>().SelectMany(t =>
-                            GetCollectionElementType(t, parameterIndex: 0));
+                            GetCollectionElementType(t));
```

Comment:
@CyrusNajmabadi This scenario seems off -- shouldn't it be looking for Add methods or something else here...? I guess I'm not quite following this one.

Reply:
Yes.  I think you're correct.  Not sure why we did things this way.